internal/cmdliveinit/cmdliveinit.go

```diff
@@ -182,54 +187,70 @@ func (c *ConfigureInventoryInfo) Run(ctx context.Context) error {
 	return nil
 }
 
-// Run fills in the inventory object values into the Kptfile.
-func updateKptfile(p *pkg.Pkg, inv *kptfilev1.Inventory, force bool) error {
-	const op errors.Op = "cmdliveinit.updateKptfile"
-	// Read the Kptfile io io.dir
+// createRGFile fills in the inventory object values into the resourcegroup object and writes to file storage.
+func createRGFile(p *pkg.Pkg, inv *kptfilev1.Inventory, filename string, force bool) error {
+	const op errors.Op = "cmdliveinit.updateResourceGroup"
+	// Read the resourcegroup object io io.dir
+	rg, err := p.ReadRGFile(filename)
+	if err != nil && !goerrors.Is(err, os.ErrNotExist) {
+		return errors.E(op, p.UniquePath, err)
+	}
+
+	// Read the Kptfile to ensure that inventory information is not in Kptfile either.
 	kf, err := p.Kptfile()
 	if err != nil {
 		return errors.E(op, p.UniquePath, err)
 	}
-	// Validate the inventory values don't already exist
+	// Validate the inventory values don't exist in Kptfile.
 	isEmpty := kptfileInventoryEmpty(kf.Inventory)
 	if !isEmpty && !force {
+		return errors.E(op, p.UniquePath, &InvInKfExistsError{})
+	}
+	// Set the Kptfile inventory to be nil since we force write to resourcegroup instead.
+	kf.Inventory = nil
+
+	// Validate the inventory values don't already exist in Resourcegroup.
+	if rg != nil && !force {
 		return errors.E(op, p.UniquePath, &InvExistsError{})
 	}
-	// Finally, set the inventory parameters in the Kptfile and write it.
-	kf.Inventory = inv
-	if err := kptfileutil.WriteFile(p.UniquePath.String(), kf); err != nil {
+	// Initialize new resourcegroup object, as rg should have been nil.
+	rg = &rgfilev1alpha1.ResourceGroup{ResourceMeta: rgfilev1alpha1.DefaultMeta}
+	// // Finally, set the inventory parameters in the ResourceGroup object and write it.
+	rg.Name = inv.Name
+	rg.Namespace = inv.Namespace
+	if inv.InventoryID != "" {
+		rg.Labels = map[string]string{rgfilev1alpha1.RGInventoryIDLabel: inv.InventoryID}
+	}
+	if err := writeRGFile(p.UniquePath.String(), rg, filename); err != nil {
 		return errors.E(op, p.UniquePath, err)
 	}
+
+	// Rewrite Kptfile if inventory exists
+	if !isEmpty {
+		if err := kptfileutil.WriteFile(p.UniquePath.String(), kf); err != nil {
+			return errors.E(op, p.UniquePath, err)
+		}
+	}
```

Comment:
Ditto. See comment above about how to do backward compatibility support.

Reply:
Per above comment, we handle backwards compatibility already. This is required if the user specifies the `--force` flag to reinitialize. This way, we write out the new inventory to `resourcegroup.yaml` and remove the old Kptfile.inventory.